eva/interfaces/relational/utils.py

```diff
@@ -0,0 +1,117 @@
+from eva.parser.select_statement import SelectStatement
+from eva.parser.statement import AbstractStatement
+from eva.parser.table_ref import TableRef
+
+from eva.parser.utils import (
+    parse_expression,
+    parse_lateral_join,
+    parse_predicate_expression,
+)
+
+
+def sql_string_to_expresssion_list(expr: str) -> List[AbstractExpression]:
+    """Converts the sql expression to list of eva abstract expressions
+
+    Args:
+        expr (str): the expr to convert
+
+    Returns:
+        List[AbstractExpression]: list of eva abstract expressions
+
+    """
+    return parse_expression(expr)
+
+
+def sql_predicate_to_expresssion_tree(expr: str) -> AbstractExpression:
+    return parse_predicate_expression(expr)
+
+
+def execute_statement(statement: AbstractStatement) -> Batch:
+    StatementBinder(StatementBinderContext()).bind(statement)
+    l_plan = StatementToPlanConverter().visit(statement)
+    p_plan = asyncio.run(PlanGenerator().build(l_plan))
+    output = PlanExecutor(p_plan).execute_plan()
+    if output:
+        batch_list = list(output)
+        return Batch.concat(batch_list, copy=False)
+
+
+def string_to_lateral_join(expr: str, alias: str):
+    return parse_lateral_join(expr, alias)
+
+
+def create_star_expression():
+    return [TupleValueExpression(col_name="*")]
+
+
+def create_limit_expression(num: int):
+    return ConstantValueExpression(num)
+
+
+def try_binding(stmt: AbstractStatement):
+    # To avoid complications in subsequent binder calls, we attempt to bind a copy of
+    # the statement since the binder modifies the statement in place and can cause
+    # issues if statement is partially bound.
+    StatementBinder(StatementBinderContext()).bind(stmt.copy())
+
+
+def handle_select_clause(
+    query: SelectStatement, alias: str, clause: str, value: Union[str, int, list]
+) -> SelectStatement:
+    """
+    Modifies a SELECT statement object by adding or modifying a specific clause.
+
+    Args:
+        query (SelectStatement): The SELECT statement object.
+        alias (str): Alias for the table reference.
+        clause (str): The clause to be handled.
+        value (str, int, list): The value to be set for the clause.
+
+    Returns:
+        SelectStatement: The modified SELECT statement object.
+
+    Raises:
+        AssertionError: If the query is not an instance of SelectStatement class.
+        AssertionError: If the clause is not in the accepted clauses list.
+    """
+
+    assert isinstance(
+        query, SelectStatement
+    ), "query must be an instance of SelectStatement"
+
+    accepted_clauses = [
+        "where_clause",
+        "target_list",
+        "groupby_clause",
+        "orderby_list",
+        "limit_count",
+    ]
+
+    assert clause in accepted_clauses, f"Unknown clause: {clause}"
+
+    # If the clause being set is "target_list" and the value is equal to
+    # "*", the "SELECT *" portion is reset.
```

Comment:
Why is it reset -- may be expand the comment?

Reply:
Done!